Extension/src/LanguageServer/configurations.ts

```diff
@@ -1530,6 +1530,11 @@ export class CppProperties {
                 result = result.replace("${workspaceRoot}", this.rootUri.fsPath);
             }
         }
+
+        if (result.includes("${userHome}")) {
```

Comment:
Please add this to the `ExtendedEnvironment` instead. (line 421-ish).

Might also want to update the regexp on line 374 of `common.ts` as well, though I'm not sure why all the variable names were added there (maybe the ones after `env|config|workspaceFolder` should be deleted instead)

Reply:
What about these other variables that are being hard-coded?  i.e. vcpkgroot.  Should they be moved into ExtendedEnvironment()/AdditionalEnvironment() ?

Reply:
If vcpkgRoot is not processed on the native side, then I think it makes sense to move it to ExtendedEnvironment.  workspaceFolder needs to stay here though because resolveVariables does not touch it and it seems we need valid paths when this function completes.

Reply:
vcpkgRoot does not seem to have any dependency on native code so I will add it to Extended Environment